tests_nccl/test_distributedarray_nccl.py

```diff
@@ -0,0 +1,412 @@
+"""Test the DistributedArray class
+Designed to run with n processes
+$ mpiexec -n 10 pytest test_distributedarray.py --with-mpi
+
+This file employs the same test sets as test_distributedarray under NCCL environment
+"""
+
+import numpy as np
+from mpi4py import MPI
+import pytest
+from numpy.testing import assert_allclose
+
+from pylops_mpi import DistributedArray, Partition
+from pylops_mpi.DistributedArray import local_split
+from pylops_mpi.utils.backend import initialize_nccl_comm, get_cupy
+
+cp = get_cupy()
```

Comment:
no need to do this in tests, tests are developer facing and we know that if we run this test we need cupy... it is only for user facing code that we want to allow one to use pylops-mpi with and without cupy/nccl

I think the same may apply to `initialize_nccl_comm` but since that is a bit more complex method that does quite a few things and may be also useful in scripts, I will keep where it is 😄